components/external-services-mock/internal/formationnotification/handler.go

```diff
@@ -232,58 +240,83 @@ func (h *Handler) RespondWithIncomplete(writer http.ResponseWriter, r *http.Requ
+
+			httputils.RespondWithBody(ctx, writer, http.StatusOK, json.RawMessage(response))
+			return
+		}
+
+		response := FormationAssignmentResponseBodyWithState{
+			State: ReadyAssignmentState,
+		}
+		httputils.RespondWithBody(ctx, writer, http.StatusOK, response)
+	}
+
+	h.syncFAResponse(ctx, writer, r, responseFunc)
 }
 
 // Delete handles synchronous formation assignment notification requests for Unassign operation
 func (h *Handler) Delete(writer http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
 	responseFunc := func([]byte) { writer.WriteHeader(http.StatusOK) }
 
-	h.syncFAResponse(writer, r, responseFunc)
+	h.syncFAResponse(ctx, writer, r, responseFunc)
 }
 
-// DeleteWithState handles synchronous formation assignment notification requests for Unassign operation and returns state in the response body
-func (h *Handler) DeleteWithState(writer http.ResponseWriter, r *http.Request) {
+// DestinationDelete handles synchronous formation assignment notification requests for destination deletion during Unassign operation
+func (h *Handler) DestinationDelete(writer http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
 	responseFunc := func([]byte) {
-		response := FormationAssignmentResponseBodyWithState{State: "READY"}
+		response := FormationAssignmentResponseBodyWithState{
+			State: ReadyAssignmentState,
+		}
 		httputils.RespondWithBody(context.TODO(), writer, http.StatusOK, response)
 	}
 
-	h.syncFAResponse(writer, r, responseFunc)
+	h.syncFAResponse(ctx, writer, r, responseFunc)
 }
 
-// GetResponses returns the notification data saved in the Mappings
-func (h *Handler) GetResponses(writer http.ResponseWriter, r *http.Request) {
-	if bodyBytes, err := json.Marshal(&h.Mappings); err != nil {
-		httphelpers.WriteError(writer, errors.Wrap(err, "body is not a valid JSON"), http.StatusBadRequest)
-		return
-	} else {
-		writer.WriteHeader(http.StatusOK)
-		_, err = writer.Write(bodyBytes)
-		if err != nil {
-			httphelpers.WriteError(writer, errors.Wrap(err, "error while writing response"), http.StatusInternalServerError)
-			return
-		}
+// DeleteWithState handles synchronous formation assignment notification requests for Unassign operation and returns state in the response body
+func (h *Handler) DeleteWithState(writer http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	responseFunc := func([]byte) {
+		response := FormationAssignmentResponseBodyWithState{State: "READY"}
```

Comment:
The handler is only moved(copied from one place and pasted into another), I didn't change anything. `DestinationDelete` is a new handler that we added and there we use `ReadyAssignmentState`. However, I'll change it here as well.
About the variable, I think it will be unecessary because we already have the structure and we build what we need. And the name will be very long because we have a couple of "response structures" - `FormationAssignmentResponseBodyWithState`, `FormationAssignmentResponseBody`, `FormationAssignmentResponseConfig`. And if the var name is "generic", e.g. `formationAssignmentReadyResponse`(or somethign similar) it will be misleading because it will look like the var could be reused for all "responses" and that wouldn't be the case